### ISSUE TITLE
"Unexpected token ILLEGAL" on document Save

### ISSUE THREAD

**Issue by matthurne** [bobthecow/genghis#33]
If I click "Edit" on a document when viewing the contents of a collection and then click "Save", sometimes I get an "Unexpected token ILLEGAL" error even if I didn't make any changes to the document. Based on the line number called out by the error, this seems to be caused by `[ ... ]` showing up where it shouldn't.

When viewing such a document in the non-edit view, there is an array field with a large number of values that is collapsed and displayed like:

```
fieldname: [ ... ],
```

When I click edit, the field is not collapsed, and is displayed like:

```
fieldname: [
    {
        anotherfield: "somevalue"
    },
    {
        anotherfield: "somevalue"
    }
][ ... ],
```

It seems to me that the `[ ... ]` shouldn't be there.

**Comment by bobthecow** [comment]
So you're seeing both the collapsed (`[ ... ]`) and uncollapsed versions of the same property?

**Comment by matthurne** [comment]
Correct, in the editor the collapsed `[ ... ]` representation is there just after the literal uncollapsed representation.

**Comment by bobthecow** [comment]
Thanks for reporting this. I've got a fix in for `v2.1`.
